freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/instance/InstanceMetaDataService.java

```diff
@@ -42,6 +43,19 @@ public void saveInstanceRequests(Stack stack, List<Group> groups) {
         }
     }
 
+    public void updateStatus(Stack stack, String instanceId, com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceStatus status) {
+        Set<InstanceGroup> instanceGroups = stack.getInstanceGroups();
+        for (InstanceGroup group : instanceGroups) {
+            for (InstanceMetaData instanceMetaData : group.getInstanceMetaData()) {
+                if (instanceId.equals(instanceMetaData.getInstanceId())) {
+                    instanceMetaData.setInstanceStatus(status);
+                    instanceMetaDataRepository.save(instanceMetaData);
+                    return;
+                }
+            }
+        }
```

Comment:
stream usage seems reasonable here

Reply:
also a log for changing the status. and unit test please

Reply:
fixed